enterprise/server/raft/filestore/filestore.go

```diff
@@ -494,16 +496,43 @@ type Store interface {
 	FileReader(ctx context.Context, fileDir string, f *rfpb.StorageMetadata_FileMetadata, offset, limit int64) (io.ReadCloser, error)
 	FileWriter(ctx context.Context, fileDir string, fileRecord *rfpb.FileRecord) (interfaces.CommittedMetadataWriteCloser, error)
 
+	BlobReader(ctx context.Context, b *rfpb.StorageMetadata_GCSMetadata, offset, limit int64) (io.ReadCloser, error)
+	BlobWriter(ctx context.Context, fileRecord *rfpb.FileRecord) (interfaces.CommittedMetadataWriteCloser, error)
+	DeleteStoredBlob(ctx context.Context, b *rfpb.StorageMetadata_GCSMetadata) error
+
 	DeleteStoredFile(ctx context.Context, fileDir string, md *rfpb.StorageMetadata) error
 	FileExists(ctx context.Context, fileDir string, md *rfpb.StorageMetadata) bool
 }
 
+type Options struct {
+	gcs     *gcs.GCSBlobStore
+	appName string
+}
+
+type Option func(*Options)
+
+func WithGCSBlobstore(gcs *gcs.GCSBlobStore, appName string) Option {
+	return func(o *Options) {
+		o.gcs = gcs
+		o.appName = appName
+	}
+}
+
 type fileStorer struct {
+	gcs     *gcs.GCSBlobStore
+	appName string
 }
 
 // New creates a new filestorer interface.
-func New() Store {
-	return &fileStorer{}
+func New(opts ...Option) Store {
```

Comment:
nit: I don't think we use option objects anywhere else in our code base. We typically stick to plain Options structs.

Reply:
I have added some :P I think this is nice because callsites that don't care about this don't have to change. LMK if you strongly object though.